app/src/main/java/org/breezyweather/theme/weatherView/materialWeatherView/implementor/MeteorShowerImplementor.kt

```diff
@@ -100,7 +100,7 @@ class MeteorShowerImplementor(
         }
 
         fun update(interval: Long, deltaRotation3D: Float) {
-            if (y > mCanvasSize) {
+            if (y > mCanvasSize * 1.5) {
```

Comment:
Where does that magic number come from? Shouldn't `mCanvasSize` be fixed instead?

Reply:
Sorry, I haven't had much time to take another look at this until now. I think what was happening was that [this calculation](https://github.com/breezy-weather/breezy-weather/blob/e1bb8713e04ccab1de18a9e7da14961a2fdd95cd/app/src/main/java/org/breezyweather/theme/weatherView/materialWeatherView/implementor/MeteorShowerImplementor.kt#L238):
```Java
canvas.drawLine(
    m.rectF.centerX(), m.rectF.top,
    m.rectF.centerX(), m.rectF.bottom,
    mPaint
)
```
uses the center of the rectangle as the starting point, and the above doesn't account for that. I'm not all that familiar with graphics programming so if that doesn't sound right just let me know!

Reply:
> Where does that magic number come from? Shouldn't `mCanvasSize` be fixed instead?

Speaking of magic numbers, would it be an issue if I cleaned up this file a bit to make some of the calculations more readable? It would mostly be formatting changes to the init functions and pulling some of the other magic numbers out into their own constants.

Reply:
The line is drawn through the center of the rectangle, but it starts at its top ( = y). So the comparison `y > mCanvasSize` should be fine.

However, `mViewWidth` which is used to calculate `mCanvasSize` is [fixed](https://github.com/breezy-weather/breezy-weather/blob/e1bb8713e04ccab1de18a9e7da14961a2fdd95cd/app/src/main/java/org/breezyweather/common/extensions/DisplayExtensions.kt#L44-L45) in landscape mode. It's initially set in [MaterialPainterView](https://github.com/breezy-weather/breezy-weather/blob/e1bb8713e04ccab1de18a9e7da14961a2fdd95cd/app/src/main/java/org/breezyweather/theme/weatherView/materialWeatherView/MaterialPainterView.kt#L199) and then passed in [WeatherImplementorFactory.getWeatherImplementor](https://github.com/breezy-weather/breezy-weather/blob/e1bb8713e04ccab1de18a9e7da14961a2fdd95cd/app/src/main/java/org/breezyweather/theme/weatherView/materialWeatherView/MaterialPainterView.kt#L332) which in turn calls `MeteorShowerImplementor`. As a result, `mCanvasSize` is smaller than the actual screen size of the device which leads to `y` being greater than `mCanvasSize` when the rectangle is actually still visible on the screen.

`mViewWidth` is also used to calculate other variables such as stroke width and speed. So it's probably not as simple as fixing only `mViewWidth`.

Reply:
That makes sense, I thought this issue might be related to those constants at first but I managed to convince myself otherwise somehow. Since that's the case, would the fix not be as simple as changing this call in MaterialPainterView:

```Java
canvasSize = intArrayOf(
    context.getTabletListAdaptiveWidth(metrics.widthPixels),
    metrics.heightPixels
)
```
to this?

```Java
canvasSize = intArrayOf(
    metrics.widthPixels,
    metrics.heightPixels
)
```

It doesn't seem to cause any issues with other types of weather on the phone or tablet I'm using for tests, but I'm not sure if there could be side effects I'm unaware of.

Reply:
Did you test the drawer?

Reply:
The drawer is unaffected by that change. Additionally, the overridden `onSizeChanged` method would need to be changed in the same way.

Reply:
As mentioned above, `mViewWidth` also influences the line's width and creates some unsually wide strokes; at least in my opinion. Not sure if that's an issue.

<details>

<summary>examples</summary>

| type | screenshot |
| --- | --- |
| fixed width | ![fixed width_stroke width](https://github.com/user-attachments/assets/1522af45-c1aa-479e-8842-077b08236ee4) | 
| device width - normal stroke width | ![device width_stroke width normal](https://github.com/user-attachments/assets/69fd284a-c2fa-498a-8912-727fec42d996) |
| device width - unsual stroke width | ![device width_stroke width unusual](https://github.com/user-attachments/assets/8a99caea-d7db-4973-86d0-f4f1f5dfb10e) |

**recordings**
emulated tablet 15" (width: 1920; height: 1080)

| fixed width | device width |
| --- | --- |
| <video src="https://github.com/user-attachments/assets/fa65155b-a383-4528-92d0-ebb353e714cb" /> | <video src="https://github.com/user-attachments/assets/73b4d357-47a2-411d-b2c7-d7e4c135c348" /> |

</details>

Reply:
> Did you test the drawer?

After testing the drawer again it actually is being affected by the change due to the measured width changing multiple times while the drawer opens or closes. That can be avoided by moving the block of code in `onSizeChanged` into the orientation listener. I pushed that change as well as some tweaks to the width and height calculations in the `Meteor` class so that everyone can be on the same page, I can revert the changes if needed. 

The other implementors are being affected as well by this change in similar ways. If this is the direction we'd like to go I can work on changing those to reflect the width change as well.